File: priv/combined_metrics/samples/variable_naming/name_is_single_letter/good/calculator.js
// Pricing and discount calculator with descriptive variable names.
// GOOD: params and locals named price, discount, quantity, amount make intent clear.

function applyDiscount(price, discountPercent) {
  const discounted = price * (1 - discountPercent / 100);
  return Math.round(discounted * 100) / 100;
}

function calculateTotal(amounts, taxRate) {
  const subtotal = amounts.reduce((acc, amount) => acc + amount, 0);
  const total = subtotal * (1 + taxRate / 100);
  return Math.round(total * 100) / 100;
}

function tieredPrice(price, tiers) {
  return tiers.reduce((currentPrice, [threshold, discountPercent]) => {
    if (currentPrice > threshold) return currentPrice * (1 - discountPercent / 100);
    return currentPrice;
  }, price);
}

function splitPayment(amount, count) {
  const installment = Math.round((amount / count) * 100) / 100;
  const lastInstallment = Math.round((amount - installment * (count - 1)) * 100) / 100;
  return [...Array(count - 1).fill(installment), lastInstallment];
}

function compoundDiscount(price, discountPercents) {
  const finalPrice = discountPercents.reduce(
    (currentPrice, discountPercent) => currentPrice * (1 - discountPercent / 100),
    price
  );
  return Math.round(finalPrice * 100) / 100;
}

function priceWithTax(unitPrice, quantity, taxRate) {
  const subtotal = unitPrice * quantity;
  const tax = subtotal * taxRate / 100;
  return {
    subtotal: Math.round(subtotal * 100) / 100,
    tax: Math.round(tax * 100) / 100,
    total: Math.round((subtotal + tax) * 100) / 100,
  };
}

function bulkPricing(price, quantity, bulkThreshold) {
  if (quantity >= bulkThreshold) return Math.round(price * 0.75 * 100) / 100;
  if (quantity >= bulkThreshold / 2) return Math.round(price * 0.9 * 100) / 100;
  return price;
}

function margin(sellingPrice, cost) {
  const marginPercent = (sellingPrice - cost) / sellingPrice * 100;
  return Math.round(marginPercent * 100) / 100;
}

function currencyConvert(amount, exchangeRate, conversionFee) {
  const converted = amount * exchangeRate;
  const afterFee = converted - converted * conversionFee / 100;
  return Math.round(afterFee * 100) / 100;
}

function installmentSchedule(principal, count, annualRate) {
  const totalAmount = principal * (1 + annualRate / 100);
  const installment = Math.round((totalAmount / count) * 100) / 100;
  return Array.from({ length: count }, (_, index) => ({
    installment: index + 1,
    amount: index < count - 1 ? installment : Math.round((totalAmount - installment * (count - 1)) * 100) / 100,
  }));
}

module.exports = { applyDiscount, calculateTotal, tieredPrice, splitPayment, compoundDiscount, priceWithTax, bulkPricing, margin, currencyConvert, installmentSchedule };

File: priv/combined_metrics/samples/variable_naming/name_is_single_letter/good/calculator.rb
# Pricing and discount calculator with descriptive variable names.
# GOOD: params and locals named price, discount, quantity, amount make intent clear.

class CalculatorGood
  def apply_discount(price, discount_percent)
    discounted = price * (1 - discount_percent / 100.0)
    discounted.round(2)
  end

  def calculate_total(amounts, tax_rate)
    subtotal = amounts.sum
    total = subtotal * (1 + tax_rate / 100.0)
    total.round(2)
  end

  def tiered_price(price, tiers)
    tiers.reduce(price) do |current_price, (threshold, discount_percent)|
      current_price > threshold ? current_price * (1 - discount_percent / 100.0) : current_price
    end
  end

  def split_payment(amount, count)
    installment = (amount / count).round(2)
    last_installment = (amount - installment * (count - 1)).round(2)
    Array.new(count - 1, installment) + [last_installment]
  end

  def compound_discount(price, discount_percents)
    final_price = discount_percents.reduce(price) do |current_price, discount_percent|
      current_price * (1 - discount_percent / 100.0)
    end
    final_price.round(2)
  end

  def price_with_tax(unit_price, quantity, tax_rate)
    subtotal = unit_price * quantity
    tax = subtotal * tax_rate / 100.0
    {
      subtotal: subtotal.round(2),
      tax: tax.round(2),
      total: (subtotal + tax).round(2)
    }
  end

  def bulk_pricing(price, quantity, bulk_threshold)
    if quantity >= bulk_threshold
      (price * 0.75).round(2)
    elsif quantity >= bulk_threshold / 2
      (price * 0.9).round(2)
    else
      price
    end
  end

  def margin(selling_price, cost)
    margin_percent = (selling_price - cost) / selling_price.to_f * 100
    margin_percent.round(2)
  end

  def currency_convert(amount, exchange_rate, conversion_fee)
    converted = amount * exchange_rate
    after_fee = converted - converted * conversion_fee / 100.0
    after_fee.round(2)
  end

  def installment_schedule(principal, count, annual_rate)
    total_amount = principal * (1 + annual_rate / 100.0)
    installment = (total_amount / count).round(2)

    (1..count).map do |index|
      payment = index < count ? installment : (total_amount - installment * (count - 1)).round(2)
      { installment: index, amount: payment }
    end
  end
end
